build_runner_core/lib/src/asset_graph/graph.dart

```diff
@@ -524,6 +524,11 @@ class AssetGraph {
           isHidden: isHidden);
       if (existing != null) {
         newNode.outputs.addAll(existing.outputs);
+        // Ensure we set up the reverse link for NodeWithInput nodes.
+        existing.outputs
+            .map(get)
+            .whereType<NodeWithInputs>()
+            .forEach((node) => node.inputs.add(output));
```

Comment:
[nit] avoid `forEach` with a function literal.

```dart
var reverseDeps = existing.outputs.map(get).whereType<NodeWithInputs>();
for(var reverseDep in reverseDeps) reversDep.inputs.add(output);
```

Reply:
Does that need to apply to vm only code? I intentionally structured this to trick that lint into not complaining lmao because I think this is a fair bit cleaner 🤣 . I am happy to change it though if you would like.

Reply:
Yeah I think it still applies. The rule isn't specific to any platform.

https://dart.dev/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal

I guess it is an "Avoid" and not a "Don't" though, so 🤷 

I am curious how you managed to trick the lint though.

Reply:
The lint doesn't trigger if you have a chain of iterables like this, it is only if you have a standalone `<iterable>.forEach`.

Reply:
Although it is pretty funny to me that to work around the lint I made it even slower 🤣

Reply:
Ok, I refactored this to call a private method so I didn't have to deal with shadowing etc which was the primary reason I had for not wanting to use a for in loop in the first place.